drivers/sensor/ntc_thermistor/ntc_thermistor_calc.c

```diff
@@ -62,29 +60,33 @@ static void ntc_lookup_comp(const struct ntc_type *type, unsigned int ohm, int *
 	*i_high = high;
 }
 
-/**
- * ntc_get_ohm_of_thermistor() - Calculate the resistance read from NTC Thermistor
- *
- * @cfg: NTC Thermistor configuration
- * @sample_mv: Measured voltage in mV
- */
-uint32_t ntc_get_ohm_of_thermistor(const struct ntc_config *cfg, int sample_mv)
+uint32_t ntc_get_ohm_of_thermistor(const struct ntc_config *cfg, int sample_value,
+				   int sample_value_max)
 {
-	int pullup_mv = cfg->pullup_uv / 1000;
 	uint32_t ohm;
 
-	if (sample_mv <= 0) {
+	if (sample_value <= 0) {
 		return cfg->connected_positive ? INT_MAX : 0;
 	}
 
-	if (sample_mv >= pullup_mv) {
+	if (sample_value >= sample_value_max) {
 		return cfg->connected_positive ? 0 : INT_MAX;
 	}
 
 	if (cfg->connected_positive) {
-		ohm = cfg->pulldown_ohm * (pullup_mv - sample_mv) / sample_mv;
+		ohm = cfg->pulldown_ohm * (sample_value_max - sample_value) / sample_value;
+		if (ohm > cfg->pullup_ohm) {
+			ohm -= cfg->pullup_ohm;
+		} else {
+			ohm = 0;
+		}
```

Comment:
Based on the documentation from `zephyr/dts/bindings/sensor/ntc-thermistor.yaml`, I was assuming that pullup resistor _and_ pulldown resistor are considered for the calculation because there is no hint that one of them is ignored by the implementation based on `connected-positive` setting.

If that resistor is static or a FET switch does not matter for a correct calculation. The handling of a FET can be done by Zephyr's power management and a `power-domain`. Every Zephyr sensor has that property.

Reply:
Thanks for your reply!

I'm not saying the calculation is wrong: I'm saying I don't think it's inituitive nor documented enough for users to properly use it.

> I suggest either we scrub this from the PR or implement it in a different way that is accessible for users.

OTOH - This appears to be a 'by-product' of the original PR. Why not handle this on a separate PR instead?

Reply:
OK, I agree. I dropped that section.

It's no longer that important to me anyway. Maybe I will find some time to write a new PR with a proper documentation.